swap/src/asb/rendezvous.rs

```diff
@@ -0,0 +1,155 @@
+use crate::rendezvous::XmrBtcNamespace;
+use anyhow::Result;
+use libp2p::core::connection::ConnectionId;
+use libp2p::identity::Keypair;
+use libp2p::multiaddr::Protocol;
+use libp2p::rendezvous::{Event, Namespace};
+use libp2p::swarm::{
+    IntoProtocolsHandler, NetworkBehaviour, NetworkBehaviourAction, NetworkBehaviourEventProcess,
+    PollParameters, ProtocolsHandler,
+};
+use libp2p::{Multiaddr, PeerId};
+use std::task::{Context, Poll};
+use std::time::{Duration, Instant};
+
+/// A `NetworkBehaviour` that handles registration of the xmr-btc swap service with a rendezvous point
+pub struct Behaviour {
+    rendezvous_behaviour: libp2p::rendezvous::Rendezvous,
+    rendezvous_point_peer_id: PeerId,
+    rendezvous_point_addr: Multiaddr,
+    rendezvous_namespace: XmrBtcNamespace,
+    rendezvous_reregister_timestamp: Option<Instant>,
+    is_connected: bool,
+    events: Vec<NetworkBehaviourAction<BehaviourInEvent, ()>>,
+}
+
+impl Behaviour {
+    pub fn new(
+        keypair: Keypair,
+        peer_id: PeerId,
+        addr: Multiaddr,
+        namespace: XmrBtcNamespace,
+    ) -> Self {
+        Self {
+            rendezvous_behaviour: libp2p::rendezvous::Rendezvous::new(
+                keypair,
+                libp2p::rendezvous::Config::default(),
+            ),
+            rendezvous_point_peer_id: peer_id,
+            rendezvous_point_addr: addr,
+            rendezvous_namespace: namespace,
+            rendezvous_reregister_timestamp: None,
+            is_connected: false,
+            events: vec![],
+        }
+    }
+
+    pub fn refresh_registration(&mut self) -> Result<()> {
```

Comment:
Is there a way to encapsulate the complete refresh within the Behaviour?
I know the behaviour is not async, but I wonder if there is a way to still handle the registrations internally instead of needing the refresh from the outside?

Reply:
we could move it into the poll function, that might be a good idea